test/app-tap/CMakeLists.txt

```diff
@@ -6,3 +6,103 @@ target_link_libraries(module_api msgpuck)
 set(CMAKE_C_FLAGS "-Wall -Wextra -std=c99")
 set(CMAKE_C_FLAGS_DEBUG "-g -O0 -Werror")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2")
+
+make_lua_path(LUA_PATH
+  PATHS
+    ./?.lua
+    ${CMAKE_CURRENT_SOURCE_DIR}/lua/?.lua
+    # Needed for test-run/test_run.lua.
+    ${PROJECT_SOURCE_DIR}/test-run/?.lua
+)
+
+make_lua_path(LUA_CPATH
+  PATHS
+  ${CMAKE_CURRENT_BINARY_DIR}/?.so
+)
+
+set(TEST_SUITE_NAME "app-tap")
+set(TARANTOOL_BIN $<TARGET_FILE:tarantool>)
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target <app-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS module_api
+)
+
+# XXX: Setting PATH is required by app-tap/minimal.test.lua.
+# I would pass TARANTOOL_BIN instead and update the test.
+set(ENV_PATH $ENV{PATH})
```

Comment:
I saw several declarations of the `ENV_PATH`. Maybe it is better to declare it once in the parent CMake scope.

Reply:
`ENV_PATH` was removed in `app-luatest` as it is not needed, in `app-tap` was a single test that required setting `PATH` to a directory with `tarantool` binary - `app-tap/minimal.test.lua`. The test was fixed and `ENV_PATH` was removed as well in `app-tap/CMakeLists.txtx`:

```diff
diff --git a/test/app-luatest/CMakeLists.txt b/test/app-luatest/CMakeLists.txt
index 3adc7204ff..49d81bacca 100644
--- a/test/app-luatest/CMakeLists.txt
+++ b/test/app-luatest/CMakeLists.txt
@@ -16,11 +16,6 @@ _add_test_suite_target(${TEST_SUITE_NAME}
   DEPENDS libyield
 )
 
-# XXX: Setting PATH is required for running tarantoolctl.
-# I would pass TARANTOOL_BIN instead and update the test.
-set(ENV_PATH $ENV{PATH})
-set(ENV_PATH "${PROJECT_BINARY_DIR}/src:${ENV_PATH}")
-
 list(APPEND TEST_ENV
   "LUA_PATH=${LUATEST_LUA_PATH}"
   "LUA_CPATH=${LUA_CPATH}"
@@ -28,7 +23,6 @@ list(APPEND TEST_ENV
   "BUILDDIR=${PROJECT_BINARY_DIR}"
   "PYTHON_EXECUTABLE=${PYTHON}"
   "TARANTOOLCTL=${PROJECT_SOURCE_DIR}/test-run/tarantoolctl"
-  "PATH=${ENV_PATH}"
 )
 
 file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *_test.lua)
diff --git a/test/app-tap/CMakeLists.txt b/test/app-tap/CMakeLists.txt
index ecc59a635e..24afe23d89 100644
--- a/test/app-tap/CMakeLists.txt
+++ b/test/app-tap/CMakeLists.txt
@@ -31,14 +31,9 @@ _add_test_suite_target(${TEST_SUITE_NAME}
   DEPENDS module_api
 )
 
-# XXX: Setting PATH is required by app-tap/minimal.test.lua.
-# I would pass TARANTOOL_BIN instead and update the test.
-set(ENV_PATH $ENV{PATH})
-set(ENV_PATH "${PROJECT_BINARY_DIR}/src:${ENV_PATH}")
 list(APPEND TEST_ENV
   "LUA_PATH=${LUA_PATH}"
   "LUA_CPATH=${LUA_CPATH}"
-  "PATH=${ENV_PATH}"
 )
 
 file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
diff --git a/test/app-tap/minimal.test.lua b/test/app-tap/minimal.test.lua
index 14ccec90c2..403606a8cd 100755
--- a/test/app-tap/minimal.test.lua
+++ b/test/app-tap/minimal.test.lua
@@ -1,5 +1,7 @@
 #!/usr/bin/env tarantool
 
+local TARANTOOL_PATH = arg[-1]
+
 print('Hello, World!')
 
 --
@@ -19,7 +21,7 @@ script:close()
 
 io.flush()
 
-os.execute("tarantool ./script-args.lua 1 2 3")
+os.execute(("%s ./script-args.lua 1 2 3"):format(TARANTOOL_PATH))
 
 --
 -- LUA_PATH and LUA_CPATH argument handling
@@ -56,7 +58,7 @@ for _, env in ipairs({
     }) do
         local cmd = table.concat({
             ("%s='%s'"):format(env[1], res[2]),
-            ('tarantool %s'):format(env[2]),
+            ('%s %s'):format(TARANTOOL_PATH, env[2]),
         }, ' ')
         local fh = io.popen(cmd)
         local rv = fh:read():gsub('-', '%%-'):gsub('+', '%%+'):gsub('?', '%%?')
```

Reply:
ENV_PATH removed